common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -121,10 +125,10 @@ public static boolean awaitInactivity(long timeout, TimeUnit unit) throws Interr
             throw new NullPointerException("unit");
         }
 
-        Thread watcherThread = ThreadDeathWatcher.watcherThread;
-        if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
-            return !watcherThread.isAlive();
+        Thread t = watcherThread.get();
+        if (t != null) {
```

Comment:
I am comfortable if we want to offer the following functionality:

- `awaitInactivity` - will block until all currently scheduled jobs are complete, with no race, also allows schedule to be called after this method is called.
- `awaitInactivityAndClose` - will block until all currently scheduled jobs are complete, with no race, all subsequent calls to watch will fail.

The "no race" part being the key :)